src/test/java/org/jabref/logic/importer/fetcher/BiodiversityLibraryTest.java

```diff
@@ -0,0 +1,169 @@
+
+@SuppressWarnings("checkstyle:EmptyLineSeparator")
+@FetcherTest
+public class BiodiversityLibraryTest {
+    private final BiodiversityLibrary fetcher = new BiodiversityLibrary();
+    private final String BASE_URL = "https://www.biodiversitylibrary.org/api3?";
+    private final String RESPONSE_FORMAT = "&format=json";
+    private final BuildInfo buildInfo = new BuildInfo();
+
+    @Test
+    public void testGetName() {
+        assertEquals("Biodiversity Heritage", fetcher.getName());
+        assertNotEquals("Biodiversity Heritage Library", fetcher.getName());
+        assertNotEquals("Biodiversity Library", fetcher.getName());
+    }
+
+    @Test
+    public void biodiversityHeritageApiKeyIsNotEmpty() {
+        BuildInfo buildInfo = new BuildInfo();
+        assertNotNull(buildInfo.biodiversityHeritageApiKey);
+    }
+
+    @Test
+    public void baseURLConstruction() throws MalformedURLException, URISyntaxException {
+
+        String baseURL = fetcher.getBaseURL().toString();
+        assertEquals(BASE_URL
+                        .concat("apikey=")
+                        .concat(buildInfo.biodiversityHeritageApiKey)
+                        .concat(RESPONSE_FORMAT),
+                baseURL);
+
+    }
+
+    @Test
+    public void getPartMetadaUrl() throws MalformedURLException, URISyntaxException {
+        String id = "1234";
+        String expected_base = (BASE_URL
+                .concat("apikey=")
+                .concat(buildInfo.biodiversityHeritageApiKey)
+                .concat(RESPONSE_FORMAT)
+                .concat("&op=GetPartMetadata&pages=f&names=f")
+                .concat("&id=")
+        );
+        String expected = expected_base.concat(id);
+
+        assertEquals(expected, fetcher.getPartMetadataURL(id).toString());
+
+        id = "4321";
+        expected = expected_base.concat(id);
+        assertEquals(expected, fetcher.getPartMetadataURL(id).toString());
+
+        id = "331";
+        expected = expected_base.concat(id);
+        assertEquals(expected, fetcher.getPartMetadataURL(id).toString());
+
+    }
+
+    @Test
+    public void getItemMetadaUrl() throws MalformedURLException, URISyntaxException {
+        String id = "1234";
+        String expected_base = (BASE_URL
+                .concat("apikey=")
+                .concat(buildInfo.biodiversityHeritageApiKey)
+                .concat(RESPONSE_FORMAT)
+                .concat("&op=GetItemMetadata&pages=f&ocr=f&ocr=f")
+                .concat("&id=")
+        );
+        String expected = expected_base.concat(id);
+
+        assertEquals(expected, fetcher.getItemMetadataURL(id).toString());
+
+        id = "4321";
+        expected = expected_base.concat(id);
+        assertEquals(expected, fetcher.getItemMetadataURL(id).toString());
+
+        id = "331";
+        expected = expected_base.concat(id);
+        assertEquals(expected, fetcher.getItemMetadataURL(id).toString());
+
+    }
+
+    @Test
+    public void jsonResultToBibEntry() {
```

Comment:
I assume, these are different tests - could you convert them to parameterizedTests? - https://www.baeldung.com/parameterized-tests-junit-5